vnext/Shared/Modules/BlobModule.cpp

```diff
@@ -0,0 +1,538 @@
+    {".jsonld", "application/ld+json"},
+    {".mid .midi", "audio/midi audio/x-midi"},
+    {".mjs", "text/javascript"},
+    {".mp3", "audio/mpeg"},
+    {".mp4", "video/mp4"},
+    {".mpeg", "video/mpeg"},
+    {".mpkg", "application/vnd.apple.installer+xml"},
+    {".odp", "application/vnd.oasis.opendocument.presentation"},
+    {".ods", "application/vnd.oasis.opendocument.spreadsheet"},
+    {".odt", "application/vnd.oasis.opendocument.text"},
+    {".oga", "audio/ogg"},
+    {".ogv", "video/ogg"},
+    {".ogx", "application/ogg"},
+    {".opus", "audio/opus"},
+    {".otf", "font/otf"},
+    {".png", "image/png"},
+    {".pdf", "application/pdf"},
+    {".php", "application/x-httpd-php"},
+    {".ppt", "application/vnd.ms-powerpoint"},
+    {".pptx", "application/vnd.openxmlformats-officedocument.presentationml.presentation"},
+    {".rar", "application/vnd.rar"},
+    {".rtf", "application/rtf"},
+    {".sh", "application/x-sh"},
+    {".svg", "image/svg+xml"},
+    {".swf", "application/x-shockwave-flash"},
+    {".tar", "application/x-tar"},
+    {".tif .tiff", "image/tiff"},
+    {".ts", "video/mp2t"},
+    {".ttf", "font/ttf"},
+    {".txt", "text/plain"},
+    {".vsd", "application/vnd.visio"},
+    {".wav", "audio/wav"},
+    {".weba", "audio/webm"},
+    {".webm", "video/webm"},
+    {".webp", "image/webp"},
+    {".woff", "font/woff"},
+    {".woff2", "font/woff2"},
+    {".xhtml", "application/xhtml+xml"},
+    {".xls", "application/vnd.ms-excel"},
+    {".xlsx", "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"},
+    {".xml", "application/xml"},
+    {".xul", "application/vnd.mozilla.xul+xml"},
+    {".zip", "application/zip"},
+    {".3gp", "video/3gpp; audio/3gpp if it doesn't contain video"},
+    {".3g2", "video/3gpp2; audio/3gpp2 if it doesn't contain video"},
+    {".7z", "application/x-7z-compressed"}}; // extensionToMime
+} // namespace
+
+namespace Microsoft::React {
+
+#pragma region BlobModule
+
+BlobModule::BlobModule(winrt::Windows::Foundation::IInspectable const &inspectableProperties) noexcept
+    : m_sharedState{std::make_shared<SharedState>()},
+      m_blobPersistor{std::make_shared<MemoryBlobPersistor>()},
+      m_contentHandler{std::make_shared<BlobWebSocketModuleContentHandler>(m_blobPersistor)},
+      m_uriHandler{std::make_shared<BlobModuleUriHandler>(m_blobPersistor)},
+      m_requestBodyHandler{std::make_shared<BlobModuleRequestBodyHandler>(m_blobPersistor)},
+      m_responseHandler{std::make_shared<BlobModuleResponseHandler>(m_blobPersistor)},
+      m_inspectableProperties{inspectableProperties} {
+  auto propId =
+      ReactPropertyId<ReactNonAbiValue<weak_ptr<IWebSocketModuleContentHandler>>>{L"BlobModule.ContentHandler"};
+  auto propBag = ReactPropertyBag{m_inspectableProperties.try_as<IReactPropertyBag>()};
+  auto contentHandler = weak_ptr<IWebSocketModuleContentHandler>{m_contentHandler};
+  propBag.Set(propId, std::move(contentHandler));
```

Comment:
Should this be in addWebSocketHandler ?

Reply:
Not really.
See Android's implementation:
```Java
  @Override
  public void addWebSocketHandler(final double idDouble) {
    final int id = (int) idDouble;

    WebSocketModule webSocketModule = getWebSocketModule("addWebSocketHandler");

    if (webSocketModule != null) {
      webSocketModule.setContentHandler(id, mWebSocketContentHandler);
    }
  }
```
That code adds the same singleton (`WebSocketModule`)'s reference as a handler for a given socket connection ID.
Because the handler will never change, it's useless to add references like that in our C++ code.
What we do in `addWebSocketHandler` is simpler:
```C++
       [contentHandler = m_contentHandler](dynamic args) {
         auto id = jsArgAsInt(args, 0);

         contentHandler->Register(id);
       }
```
We only keep track of the connection ID, for the object that will handle it is set for both `BlobModule` and the rest of the modules in the constructor by adding it to the property bag.